### PR TITLE
auto generate cron string for period job

### PR DESCRIPTION
**Pull request by chaodaiG**
Generate cron strings based on job type and job name, avoid the headache of managing them manually

### CODE REVIEW

File: ci/prow/make_config.go [hunk 1]
```diff
@@ -264,6 +265,51 @@ const (
...
+	// sums the ascii valus of all letters in a jobname,
+	// this value is used for deriving offset after hour
+	var sum float64
+	for _, c := range jobName + jobType {
+		sum += float64(c)
+	}
+	minutesOffset := int(math.Mod(sum, 60))
+	everyHourCron := fmt.Sprintf("%d * * * *", minutesOffset)
+	everyOtherHourCron := fmt.Sprintf("%d */2 * * *", minutesOffset)
+	everydayCron := fmt.Sprintf("%d %%d * * *", minutesOffset)    // hour
```
Comment (Fredy-Z) [inline ci/prow/make_config.go]: This looks nice!
Just one question, what does `%%d` mean here?

Comment (adrcunha) [reply ci/prow/make_config.go]: It's the literal formatting for the real value. See line 304 for example.

File: ci/prow/make_config.go [hunk 2]
```diff
@@ -264,6 +265,51 @@ const (
...
+func generateCron(jobType, jobName string) string {
+	getUTCtime := func(i int) int { return i + 7 }
+
+	// sums the ascii valus of all letters in a jobname,
+	// this value is used for deriving offset after hour
+	var sum float64
+	for _, c := range jobName + jobType {
+		sum += float64(c)
+	}
+	minutesOffset := int(math.Mod(sum, 60))
```
[the remark this answers is not in the document]
Comment (chaodaiG) [reply ci/prow/make_config.go]: Can't think about a very clean way of doing this, but I can give it a shot

File: ci/prow/make_config.go [hunk 3]
```diff
@@ -264,6 +265,51 @@ const (
...
+	everyHourCron := fmt.Sprintf("%d * * * *", minutesOffset)
+	everyOtherHourCron := fmt.Sprintf("%d */2 * * *", minutesOffset)
+	everydayCron := fmt.Sprintf("%d %%d * * *", minutesOffset)    // hour
+	everyweekCron := fmt.Sprintf("%d %%d * * %%d", minutesOffset) // hour, weekday
+
+	var res string
+	switch jobType {
+	case "continuous": // Every hour
+		res = fmt.Sprintf(everyHourCron)
```
[the remark this answers is not in the document]
Comment (chaodaiG) [reply ci/prow/make_config.go]: Good idea, done

File: ci/prow/make_config.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (chaodaiG) [reply ci/prow/make_config.go]: Make quite some change to have this work, please review again @adrcunha

**Pull request merged**